android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java

```diff
@@ -168,4 +189,242 @@ private void ensureInitialized() {
             }
         }
     }
+
+    @Override
+    public JsonElement get(String key) {
+        ensureInitialized();
+
+        if (key.equals("confirmations") || key.equals("contaminations")) {
+            LOGGER.error("[Deprecation] Retrieving confirmations and contaminations from the Evolv context with keys \"confirmations\"," +
+                    "and \"contaminations\" is deprecated. Please use \"experiments.confirmations\" and \"experiments.contaminations\" instead.");
+        }
+// TODO: 27.07.2021 question: JS SDK - "return objects.getValueForKey(key, remoteContext) || objects.getValueForKey(key, localContext);"
+//  need to understand which cases should return value local or remote (if local and remote contexts aren't emprty)
+
+        if(localContext.size() != 0){
+            return helper.getValueForKey(key, localContext).getAsJsonObject();
+        }
+
+
+        JsonElement element = helper.getValueForKey(key, remoteContext);
+        return element;
+
+    }
+
+    // TODO: 27.07.2021 need to test!
+    @Override
+    public boolean pushToArray(String key, String value, boolean local) {
```

Comment:
This code duplicates the other pushToArray

Just call pushToArray(key, value, local, DEFAULT_QUEUE_LIMIT)

Reply:
if I leave one method with four parameters **(key, value, local, DEFAULT_QUEUE_LIMIT)** then we must pass it and cannot be skipped as js and then check for underfind.
**first option:** 
if I don't want to pass a QUEUE_LIMIT, I put 0 and then there will be one method in the evolv client: **pushToArray(key, value, local, 0)**
**second option:** 
if I don't want to pass a QUEUE_LIMIT, I will choose a method that does not have this parameter (QUEUE_LIMIT) and use it **pushToArray(key, value, local)**
and then there will be two overloaded methods with different set of parameters 
pushToArray(key, value, local)
pushToArray(key, value, local,QUEUE_LIMIT)